Kernel/Time/HardwareTimer.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <Kernel/Devices/Timers/HardwareTimer.h>
+#include <Kernel/Devices/Timers/TimeManagement.h>
+
+namespace Kernel {
+
+HardwareTimer::HardwareTimer(u8 irq_number, void (*callback)(const RegisterState&))
+    : IRQHandler(irq_number)
+{
+    ASSERT(callback != nullptr);
+    m_function_to_call = callback;
+}
+void HardwareTimer::handle_irq(const RegisterState& regs)
+{
+    m_function_to_call(regs);
+}
+const char* HardwareTimer::purpose() const
+{
+    if (TimeManagement::the().is_system_timer(*this))
+        return "System Timer";
+    return model();
+}
+void HardwareTimer::change_function(void (*new_callback)(const RegisterState&))
```

Comment:
I'm pretty sure that I'm doing something bad here... I'm not very familiar with `Function` container, so any help is appreciated.